lib/mini_mime.rb

```diff
@@ -85,14 +85,40 @@ def fetch(key, &blk)
       end
     end
 
+    if ::File.method_defined?(:pread)
+      PReadFile = ::File
+    else
+      # For Windows support
+      class PReadFile
+        def initialize(filename)
+          @mutex = Mutex.new
+          # We must open the file in binary mode
+          # otherwise Ruby's automatic line terminator
+          # translation will skew the row size
+          @file = ::File.open(filename, 'rb')
```

Comment:
Ok, so it's because for IOs in text mode, Ruby automatically replace `\r\n` by `\n`, so the `row_length` end up being off by one.